collector.go

```diff
@@ -67,21 +67,28 @@ type CollectorEndpointResponse struct {
 }
 
 func (c *Collector) Collect(ctx context.Context) error {
-	var responses [][]byte
+	c.logger.Log("level", "debug", "message", "collector starts collecting version bundles from endpoints")
+
+	responses := map[string][]byte{}
 	{
 		var g errgroup.Group
-		responses = make([][]byte, len(c.endpoints))
 
-		for i, e := range c.endpoints {
-			i, e := i, e
+		for _, endpoint := range c.endpoints {
+			e := endpoint
 
 			g.Go(func() error {
+				c.logger.Log("endpoint", e.String(), "level", "debug", "message", "collector requesting version bundles from endpoint")
+
 				res, err := c.restClient.NewRequest().Get(e.String())
 				if err != nil {
 					return microerror.Mask(err)
 				}
 
-				responses[i] = res.Body()
+				c.logger.Log("endpoint", e.String(), "level", "debug", "message", "collector received version bundles from endpoint")
+
+				c.mutex.Lock()
+				responses[e.String()] = res.Body()
+				c.mutex.Unlock()
```

Comment:
I am lazy here. There is a nicer support for concurrent maps in the sync package but I was a lazy cunt here now. We do not really need all the fancy map stuff and this just does the trick without any other changes.